### PR TITLE
fix(link-list-section): replaced wrong heading component in storybook

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#4844

### Description

Replaced the previous `<dds-content-section-heading>`component within the Link List Section story to `<dds-link-list-heading>`. 

Also noted that `<dds-content-section-heading>` didn't have the `heading` role and included that in this PR.

### Changelog

**New**

- added `role='heading'` to `<dds-content-section-heading>`

**Changed**

- replaced wrong heading component for Link List Section within storybook

### CODE REVIEW

File: packages/web-components/src/components/content-section/content-section-heading.ts [hunk 1]
```diff
@@ -23,6 +23,13 @@ class DDSContentSectionHeading extends LitElement {
   @property({ reflect: true })
   slot = 'heading';
 
+  connectedCallback() {
+    if (!this.hasAttribute('role')) {
+      this.setAttribute('role', 'heading');
+    }
```
Comment (asudoh) [inline packages/web-components/src/components/content-section/content-section-heading.ts:29]: Could you add `aria-level`, too? Here's an [example](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.15.0-rc.0/packages/web-components/src/components/card/card-heading.ts#L32-L34) for one representing `<h3>`.

**Review by asudoh — COMMENTED** [review]
Thank you for doing this @IgnacioBecerra!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @IgnacioBecerra for the update!

**Comment by jeffchew** [comment]
@IgnacioBecerra can you check why one of the CI checks are failing?

**Pull request merged**